pht_federated/aggregator/services/discovery/object_list_transform.py

```diff
@@ -0,0 +1,154 @@
+from pht_federated.aggregator.schemas.dataset_statistics import *
+from operator import itemgetter as i
+from functools import cmp_to_key
+from pht_federated.aggregator.services.discovery import statistics
+
+
+def object_to_list(dataset_statistics_object: DatasetStatistics) -> list:
+    """
+    Transforms a DatasetStatistics object to python list of integers that contain all numerical values of the object
+    :param dataset_statistics_object: DatasetStatistics
+    :return: list of integers
+    """
+
+    object_list_int = []
+
+    stats_dict = dataset_statistics_object.dict()
+
+    #print("Dataset Statistics Object : {}".format(stats_dict))
+
+    if "item_count" in stats_dict:
+        object_list_int.append(stats_dict["item_count"])
+
+    if "feature_count" in stats_dict:
+        object_list_int.append(stats_dict["feature_count"])
+
+    stats_columns = stats_dict["column_information"]
+    stats_columns_sorted = multikeysort(stats_columns, ["type", "title"])
+
+    for column in stats_columns_sorted:
+        if column["type"] == "categorical":
+            if "not_na_elements" in column:
+                object_list_int.append(column["not_na_elements"])
+            if "number_categories" in column:
+                object_list_int.append(column["number_categories"])
+            if "frequency" in column:
+                object_list_int.append(column["frequency"])
+        elif column["type"] == "numeric":
+            if "not_na_elements" in column:
+                object_list_int.append(column["not_na_elements"])
+            if "mean" in column:
+                object_list_int.append(column["mean"])
+            if "std" in column:
+                object_list_int.append(column["std"])
+            if "min" in column:
+                object_list_int.append(column["min"])
+            if "max" in column:
+                object_list_int.append(column["max"])
+        elif column["type"] == "unique":
+            if "number_of_duplicates" in column:
+                object_list_int.append(column["number_of_duplicates"])
+        elif column["type"] == "equal":
+            None
+        elif column["type"] == "unstructured":
+            if "not_na_elements" in column:
+                object_list_int.append(column["not_na_elements"])
+            if "number_targets" in column:
+                object_list_int.append(column["number_targets"])
+            if "frequency" in column:
+                object_list_int.append(column["frequency"])
```

Comment:
split into separate function i.e. `process_column(list, column)`. Which should maybe be split into sub functions for each column type

Reply:
comparision could be made against an (class) enum, to avoid typos in string literals ^^